Controllers/AdminController.cs

```diff
@@ -29,21 +29,48 @@ public ActionResult Search(SearchVM searchVM)
             {
                 //Filter the collection by Title which "contains" the given string
                 foundBooks = foundBooks
-                             .Where(b => b.Title.Contains(searchVM.Title));
+                             .Where(b => b.Title.Contains(searchVM.Title))
                 // TODO: Order the results by Title
+                            .OrderBy(b => b.Title);
             }
 
             //TODO: Add similar logic to filter foundbooks collection by last part of the Author's Name, if given
             // (HINT: consider the EndsWith() method, also adjust the Search View and ViewModel to add items)
+            if (searchVM.AuthorLastName != null)
+            {
+                foundBooks = foundBooks
+                              .Where(b => b.Author.EndsWith(searchVM.AuthorLastName));
+
+            }
 
             //TODO: Filter the collection by price between a low and high value, if given
             //       order results by descending price 
             // (Note: you will need to adjust the Search ViewModel and View to add search fields)
+            if (searchVM.MinPrice != 0)
+            {
+                foundBooks = foundBooks
+                            .Where(b => b.Price > searchVM.MinPrice)
+                            .OrderByDescending(b => b.Price);
+            }
+
+            if (searchVM.MaxPrice != 0)
+            {
+                foundBooks = foundBooks
+                            .Where(b => b.Price < searchVM.MinPrice)
+                            .OrderByDescending(b => b.Price);
+            }
+
 
             //TODO:  Create a projection as a new Book collection with the "Half-Off Sale" Price for books over $90
             //       You only need to include the Title, Author, and sale price in the projection
-            if (searchVM.HalfPriceSale) { 
-                
+            if (searchVM.HalfPriceSale) {
+                foundBooks = foundBooks
+                            .Where(b => b.Price > 90)
+                            .Select(b => new {
```

Comment:
Shelby - looks like you did not push the change we discussed in class. Please resubmit once complete